### DISCUSSION TITLE
Module not found error after updating

### DISCUSSION THREAD

**Discussion by Pablodefelipe** [ImperialCollegeLondon/sharpy#182] [Q&A] [answered]
Dear developer team,

After updating to version 1.3 using git commands, I re compiled the UVLM and xbeam libraries as well as setting the sharpy environment and sourcing the sharpy variables. For reference here is some background on where and why I am using SHARPy:
OS:  Ubunut 20.04 running via virtual box
Use: Final Year Project at Imperial College London
The error log I am getting from the unit test is copy pasted at the end.
I also tried to run one of the jupyter notebook examples:'linear_goland_flutter' and I manage to get to step 17 where the sharpy program is called. I attach the error as a screenshot. 

I am not sure if this is enough information to be able to give a possible solution, I was thinking of deleting and re-installing, as I had sharpy working before I updated it.

Many thanks!
-
![error_2021_12_10](https://user-images.githubusercontent.com/88040125/145577690-da5728b9-e758-4657-b546-68971b622c6e.PNG)
--------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
~/sharpy/sharpy/generators/__init__.py in <module>
     24     try:
---> 25         generator_interface.generators[file] = importlib.import_module(import_path + "." + file)
     26     except ModuleNotFoundError:

~/anaconda3/envs/sharpy_env/lib/python3.7/importlib/__init__.py in import_module(name, package)
    126             level += 1
--> 127     return _bootstrap._gcd_import(name[level:], package, level)
    128 

~/anaconda3/envs/sharpy_env/lib/python3.7/importlib/_bootstrap.py in _gcd_import(name, package, level)

~/anaconda3/envs/sharpy_env/lib/python3.7/importlib/_bootstrap.py in _find_and_load(name, import_)

~/anaconda3/envs/sharpy_env/lib/python3.7/importlib/_bootstrap.py in _find_and_load_unlocked(name, import_)

~/anaconda3/envs/sharpy_env/lib/python3.7/importlib/_bootstrap.py in _load_unlocked(spec)

~/anaconda3/envs/sharpy_env/lib/python3.7/importlib/_bootstrap_external.py in exec_module(self, module)

~/anaconda3/envs/sharpy_env/lib/python3.7/importlib/_bootstrap.py in _call_with_frames_removed(f, *args, **kwds)

~/sharpy/sharpy/generators/floatingforces.py in <module>
      6 from scipy.interpolate import interp1d
----> 7 from control import forced_response, TransferFunction
      8 

ModuleNotFoundError: No module named 'control'

During handling of the above exception, another exception occurred:

ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-17-3ae7ae3694f5> in <module>
----> 1 sharpy.sharpy_main.main(['', ws.route + ws.case_name + '.sharpy'])

~/sharpy/sharpy/sharpy_main.py in main(args, sharpy_input_dict)
     41     import sharpy.solvers
     42     import sharpy.postproc
---> 43     import sharpy.generators
     44     import sharpy.controllers
     45     # ------------

~/sharpy/sharpy/generators/__init__.py in <module>
     25         generator_interface.generators[file] = importlib.import_module(import_path + "." + file)
     26     except ModuleNotFoundError:
---> 27         generator_interface.generators[file] = importlib.import_module('sharpy.' + import_path + "." + file)
     28 

~/anaconda3/envs/sharpy_env/lib/python3.7/importlib/__init__.py in import_module(name, package)
    125                 break
    126             level += 1
--> 127     return _bootstrap._gcd_import(name[level:], package, level)
    128 
    129 

ModuleNotFoundError: No module named 'sharpy.sharpy'

**Comment by ngoiz** [comment, marked as answer]
Hi Pablo,

You'll need to update the conda environment to include those modules that are now used in 1.3

`conda env update -f <path_to_environment_file>` should hopefully fix it.

I added an explicit note to the release notes.

Thanks,
Norberto

**Reply by Pablodefelipe** [reply]
Hi Norberto,
After updating the environment it works!
Thank you very much!!

Best,
Pablo

**Reply by 1749335863** [reply]
![image](https://user-images.githubusercontent.com/65592096/165005123-e5a1ab61-decf-49c4-ae5c-c32071f3814d.png)
Hi，Ngoiz，
I tried to update my conda environment, but it told me "file not found" as shown in the picture. Can you help me?

**Reply by ngoiz** [reply]
Hi, you are not updating the environment correctly. `<path_to_environment_file>` as shown in the answer refers to the environment file that comes with sharp, i.e. `utils/environment_linux.yaml` (or as appropriate)

Thanks,
Norberto

**Reply by alee269** [reply]
Hi, Ngoiz, I am using the 2.0 version right now but I have the same error also after updating the environment as you suggested.
![image](https://user-images.githubusercontent.com/129768983/230084936-7b736ae0-5f22-40d3-b214-8bdc7afadc21.png)
Thank you in advance.

**Reply by sduess** [reply]
Dear alee269,

Your problem is discussed in this issue #227. The solution to your particular problem is to update Python to version 3.10 on your machine. 

Thanks,
Steffi

**Reply by alee269** [reply]
Thank you, it worked. Anyway I have a new type of error after the same command. Could you please help me? 
![Screenshot from 2023-04-07 20-17-54](https://user-images.githubusercontent.com/129768983/230930365-5923ff11-4459-45d5-b6ca-ffd4ce3a9872.png)
Thank you.

Alessandra

**Comment by aeroarun** [comment]
Try using environment_new.yml as it contains the latest python 3.10
requirement.

**Reply by alee269** [reply]
Thank you, it worked. Anyway I have a new type of error after the same command. 
![Screenshot from 2023-04-07 20-17-54](https://user-images.githubusercontent.com/129768983/230666055-c314560f-0e8d-46c3-8705-45a85e50ce3a.png)
I opened it from this path: /sharpy/docs/source/content/example_notebooks
Can you please help me? Thank you in advance.
